lib/phoenix_live_view.ex

```diff
@@ -1694,19 +1753,57 @@ defmodule Phoenix.LiveView do
         {:noreply, stream_delete_by_dom_id(socket, :songs, dom_id)}
       end
   '''
-  def stream_delete_by_dom_id(socket, name, id) do
+  def stream_delete_by_dom_id(%Socket{} = socket, name, id) do
     update_stream(socket, name, &LiveStream.delete_item_by_dom_id(&1, id))
   end
 
-  defp assign_stream(socket, name, %LiveStream{} = stream) do
-    Phoenix.Component.update(socket, :streams, fn streams ->
-      streams
-      |> Map.put(name, stream)
-      |> Map.update!(:__changed__, &MapSet.put(&1, name))
-    end)
+  defp assign_stream(%Socket{} = socket, name, items, opts) do
+    streams = socket.assigns.streams
+
+    case streams do
+      %{^name => %LiveStream{}} ->
+        new_socket =
+          Enum.reduce(items, socket, fn item, acc -> stream_insert(acc, name, item, opts) end)
+
+        if opts[:reset] do
```

Comment:
in this case it doesn't matter because reset is only a flag we throw onto the stream. But I agree it would be clearer to reset first then insert since that follows the logical order of what we do on the client.